python/rikai/spark/functions/vision.py

```diff
@@ -47,9 +49,17 @@
 
 
 @udf(returnType=ImageType())
-def image(uri: str) -> Image:
-    """Build an :py:class:`Image` from a URI."""
-    return Image(uri)
+def image(image: Union[bytes, bytearray, IOBase, str, Path]) -> Image:
+    """Build an :py:class:`Image` from
+    bytes, file-like object, str, or :py:class:`~pathlib.Path`."""
+    data, uri = None, None
+    if isinstance(image, IOBase):
```

Comment:
can you add 3 unit tests for each of the 3 branches in this if-else here?

Reply:
got it but I eliminated `IOBase` since we don't have a spark column of this type.